### ISSUE TITLE
ES5 support seems to be dropped?

### ISSUE THREAD

**Issue by erangakm** [runtypes/runtypes#243]
Hi,

I've got a project that's strict checking es5 support (with https://www.npmjs.com/package/es-check) and recently it's been tripping up on [runtypes' unicode regex](https://github.com/pelotom/runtypes/blob/master/src/types/dictionary.ts#L97).

I still see that the project's `tsconfig.json` has `es5` for target and I'm just wondering if this was intentional.

Cheers.

**Comment by yuhr** [comment]
Hm, I was just unsure if we should still stick to ES5. If so, it feels like we also need to integrate Babel transpilation into our build process, because it's hard for us to maintain legacy-compliant code on our own. @pelotom, how should we do?

IMAO most people are using Babel universally for actual applications. This is purely a question, wouldn't it be an option to use Babel for your project? Then what reasons?

**Comment by mscharley** [comment]
I work with @erangakm . Yes, it'd be an option and that was going to be our fallback, but we aren't already using Babel as the application itself is using TypeScript so we already have an new ES -> old ES transpilation for our own code in that. We'd only need Babel for supporting libraries using newer features and runtypes was the first one we use that have caused issues there.

**Comment by yuhr** [comment]
I see. Personally I don't really have much opposition to maintain ES5 support with Babel in this package's side, but I'm afraid of introducing some performance drop and/or being hit inconsistent behaviour to the web standard (especially when working with `Symbol`s) by doing that, though maybe these are small things to worry about.

So... at this point I'd just recommend using Babel in applications' side. But I still want to seek some advices for this.

**Comment by mscharley** [comment]
Yeah, similarly we don't have any real strong opinions either way. We can setup babel if we need to, we just weren't sure what the intention was here as the build system for the library seemed to be targeting ES5 but the code wasn't actually compatible so we wanted to raise it upstream to see what the current stance actually is.

**Comment by rraval** [comment]
Hi, just adding another data point.

We have a real application that wants to use runtypes but we don't use babel (our first party code uses Typescript itself to target ES5). It would be nice if this library were ES5 compliant for IE11 support, but we can look at integrating babel into our build pipeline as well.

**Comment by yuhr** [comment]
Just investigating how we could achieve optimal results, I found https://github.com/developit/microbundle is a good way to go. Integrating a bundler will also result in single file build, that resolves #175, #155, and #145 as well.

**Comment by yuhr** [comment]
Transpiling to specific target should be done on the users' side.
